astropy/io/fits/hdu/compressed.py

```diff
@@ -1519,12 +1519,16 @@ def header(self):
         if 'ZSIMPLE' in self._header:
             image_header.set('SIMPLE', self._header['ZSIMPLE'],
                              self._header.comments['ZSIMPLE'], before=0)
-        elif 'ZTENSION' in self._header:
+        if 'ZTENSION' in self._header:
             if self._header['ZTENSION'] != 'IMAGE':
                 warnings.warn("ZTENSION keyword in compressed "
                               "extension != 'IMAGE'", AstropyUserWarning)
             image_header.set('XTENSION', 'IMAGE',
                              self._header.comments['ZTENSION'], before=0)
+        elif 'ZIMAGE' in self._header:
+            if self._header['ZIMAGE'] is True:
+                image_header.set('XTENSION', 'IMAGE',
+                                 self._header.comments['ZIMAGE'], before=0)
```

Comment:
If `"ZIMAGE"` value is either a Boolean or not in the header at all then the code can be simplified:
```suggestion
        elif self._header.get("ZIMAGE"):
            image_header.set('XTENSION', 'IMAGE',
                             self._header.comments['ZIMAGE'], before=0)
```